### PR TITLE
upstream merge 15/11

### PR DESCRIPTION
**Pull request by deltanedas**
## Notable additions
- all borgs are made by specializing generic borgs
  security borgs no longer take 2 plasteel and a vest to make as a result (the vest is meaningless since you just tell the borg to get it and wait a few minutes, the secfab can even print them so you couldn't run out. in general though borgs could do with extra steps to make, shouldn't be a 5 second job)
  mediborg job still works though, gumball and all
- jumpsuits have a bit more detail
- labeller changed :trollface:
- when adding a note you set it to expire in a duration instead of at a date

## Notable exclusions
- new outpost section ruin, #1590

## Media
this took a decent bit to get working :trollface:

**Changelog**
:cl:
- add: Merged upstream changes from the last 2 weeks.
- tweak: Security borgs have baton, seclite and holoprojector modules when a chassis switches to it.
- remove: Removed the security borg wood baton module, they get real batongs now.

### CODE REVIEW

File: Content.Shared/CCVar/CCVars.Atmos.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+namespace Content.Shared.CCVar;
+
+public sealed partial class CCVars
+{
+    /// <summary>
+    ///     Whether gas differences will move entities.
+    /// </summary>
+    public static readonly CVarDef<bool> SpaceWind =
+        CVarDef.Create("atmos.space_wind", false, CVar.SERVERONLY);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Atmos.cs:11]: trolled
set it in the deltav.toml to true

File: Content.Shared/CCVar/CCVars.Vote.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    ///     Config for when the restart vote should be allowed to be called based on percentage of ghosts.
+    /// </summary>
+    public static readonly CVarDef<int> VoteRestartGhostPercentage =
+        CVarDef.Create("vote.restart_ghost_percentage", 55, CVar.SERVERONLY);
+
+    /// <summary>
+    ///     See vote.enabled, but specific to preset votes
+    /// </summary>
+    public static readonly CVarDef<bool> VotePresetEnabled =
+        CVarDef.Create("vote.preset_enabled", true, CVar.SERVERONLY);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Vote.cs:35]: this should be false in the config

File: Content.Shared/CCVar/CCVars.Shuttle.cs [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    ///     Whether the arrivals terminal should be on a planet map.
+    /// </summary>
+    public static readonly CVarDef<bool> ArrivalsPlanet =
+        CVarDef.Create("shuttle.arrivals_planet", true, CVar.SERVERONLY);
+
+    /// <summary>
+    ///     Whether the arrivals shuttle is enabled.
+    /// </summary>
+    public static readonly CVarDef<bool> ArrivalsShuttles =
+        CVarDef.Create("shuttle.arrivals", true, CVar.SERVERONLY);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Shuttle.cs:26]: pro

File: Content.Shared/CCVar/CCVars.Shuttle.cs [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    ///     Is the emergency shuttle allowed to be early launched.
+    /// </summary>
+    public static readonly CVarDef<bool> EmergencyEarlyLaunchAllowed =
+        CVarDef.Create("shuttle.emergency_early_launch_allowed", false, CVar.SERVERONLY);
+
+    /// <summary>
+    ///     How long the emergency shuttle remains docked with the station, in seconds.
+    /// </summary>
+    public static readonly CVarDef<float> EmergencyShuttleDockTime =
+        CVarDef.Create("shuttle.emergency_dock_time", 180f, CVar.SERVERONLY);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Shuttle.cs:118]: 240

File: Content.Shared/CCVar/CCVars.Shuttle.cs [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    /// </summary>
+    public static readonly CVarDef<float> EmergencyShuttleAuthorizeTime =
+        CVarDef.Create("shuttle.emergency_authorize_time", 10f, CVar.SERVERONLY);
+
+    /// <summary>
+    ///     The minimum time for the emergency shuttle to arrive at centcomm.
+    ///     Actual minimum travel time cannot be less than <see cref="ShuttleSystem.DefaultArrivalTime"/>
+    /// </summary>
+    public static readonly CVarDef<float> EmergencyShuttleMinTransitTime =
+        CVarDef.Create("shuttle.emergency_transit_time_min", 60f, CVar.SERVERONLY);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Shuttle.cs:143]: 90

File: Content.Shared/CCVar/CCVars.Shuttle.cs [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    ///     ex. a call time of 10min and turning point of 0.5 means the shuttle cannot be recalled after 5 minutes.
+    /// </summary>
+    public static readonly CVarDef<float> EmergencyRecallTurningPoint =
+        CVarDef.Create("shuttle.recall_turning_point", 0.5f, CVar.SERVERONLY);
+
+    /// <summary>
+    ///     Time in minutes after round start to auto-call the shuttle. Set to zero to disable.
+    /// </summary>
+    public static readonly CVarDef<int> EmergencyShuttleAutoCallTime =
+        CVarDef.Create("shuttle.auto_call_time", 90, CVar.SERVERONLY);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Shuttle.cs:168]: 120

File: Content.Shared/CCVar/CCVars.Shuttle.cs [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+    /// <summary>
+    ///     Time in minutes after the round was extended (by recalling the shuttle) to call
+    ///     the shuttle again.
+    /// </summary>
+    public static readonly CVarDef<int> EmergencyShuttleAutoCallExtensionTime =
+        CVarDef.Create("shuttle.auto_call_extension_time", 45, CVar.SERVERONLY);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Shuttle.cs:175]: 30

File: Content.Shared/CCVar/CCVars.Misc.cs [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    /// </summary>
+    public static readonly CVarDef<bool> ProcgenPreload =
+        CVarDef.Create("procgen.preload", true, CVar.SERVERONLY);
+
+    /// <summary>
+    ///     Enabled: Cloning has 70% cost and reclaimer will refuse to reclaim corpses with souls. (For LRP).
+    ///     Disabled: Cloning has full biomass cost and reclaimer can reclaim corpses with souls. (Playtested and balanced for MRP+).
+    /// </summary>
+    public static readonly CVarDef<bool> BiomassEasyMode =
+        CVarDef.Create("biomass.easy_mode", true, CVar.SERVERONLY);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Misc.cs:22]: false

File: Content.Shared/CCVar/CCVars.Ic.cs [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    ///     Restricts IC character names to alphanumeric chars.
+    /// </summary>
+    public static readonly CVarDef<bool> RestrictedNames =
+        CVarDef.Create("ic.restricted_names", true, CVar.SERVER | CVar.REPLICATED);
+
+    /// <summary>
+    ///     Allows flavor text (character descriptions)
+    /// </summary>
+    public static readonly CVarDef<bool> FlavorText =
+        CVarDef.Create("ic.flavor_text", false, CVar.SERVER | CVar.REPLICATED);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Ic.cs:17]: lrp just a week away

File: Content.Shared/CCVar/CCVars.Ic.cs [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+    /// <summary>
+    ///     Adds a period at the end of a sentence if the sentence ends in a letter.
+    /// </summary>
+    public static readonly CVarDef<bool> ChatPunctuation =
+        CVarDef.Create("ic.punctuation", false, CVar.SERVER);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Ic.cs:23]: awful

File: Content.Shared/CCVar/CCVars.Admin.Rules.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace Content.Shared.CCVar;
+
+public sealed partial class CCVars
+{
+    /// <summary>
+    ///     Time that players have to wait before rules can be accepted.
+    /// </summary>
+    public static readonly CVarDef<float> RulesWaitTime =
+        CVarDef.Create("rules.time", 45f, CVar.SERVER | CVar.REPLICATED);
```
Comment (MilonPL) [inline Content.Shared/CCVar/CCVars.Admin.Rules.cs:11]: 60

**Review by MilonPL — CHANGES_REQUESTED** [review]

File: Content.Shared/CCVar/CCVars.Atmos.cs (hunk 1 above)
Comment (deltanedas) [reply Content.Shared/CCVar/CCVars.Atmos.cs:11]: i love these random cvars being changed in c# for no reason WITH NO COMMENTS

Comment (MilonPL) [reply Content.Shared/CCVar/CCVars.Atmos.cs:11]: REAL

File: Content.Client/Silicons/Borgs/BorgSwitchableTypeSystem.cs [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+        BorgTypePrototype prototype)
+    {
+        if (TryComp(entity, out SpriteComponent? sprite))
+        {
+            // DeltaV: Begin sprite path stuff
+            if (prototype.SpritePath is {} path && _resourceCache.TryGetResource(TextureRoot / path, out RSIResource? resource))
+                sprite.BaseRSI = resource.RSI;
+            // DeltaV: End sprite path stuff
+            sprite.LayerSetState(BorgVisualLayers.Body, prototype.SpriteBodyState);
+            sprite.LayerSetState(BorgVisualLayers.LightStatus, prototype.SpriteToggleLightState);
```
Comment (deltanedas) [inline Content.Client/Silicons/Borgs/BorgSwitchableTypeSystem.cs]: @MilonPL need help with this

if it sets the rsi first then the layers states are wrong
if it sets the layers first they point to invalid sprite

should i just make secborg be security.rsi and use robot* states so it Just Works?

Comment (MilonPL) [reply Content.Client/Silicons/Borgs/BorgSwitchableTypeSystem.cs]: hold on lemme look into it

Comment (MilonPL) [reply Content.Client/Silicons/Borgs/BorgSwitchableTypeSystem.cs]: I'M SCREAMING INTO THE VOID AND IT'S NOT LISTENING
i ran out of fucking ideas, it doesn't want to work,
this is *an* attempt but it's a giant fucking meme
sort of works but I think the Light layer is wrong? not sure if it works correctly but you can test it yourself
```cs
    protected override void UpdateEntityAppearance(
        Entity<BorgSwitchableTypeComponent> entity,
        BorgTypePrototype prototype)
    {
        if (!TryComp(entity, out SpriteComponent? sprite))
            return;

        sprite.LayerMapRemove(BorgVisualLayers.Body);
        sprite.LayerMapRemove(BorgVisualLayers.LightStatus);
        sprite.LayerMapRemove(BorgVisualLayers.Light);

        for (var i = sprite.AllLayers.Count() - 1; i >= 0; i--)
        {
            sprite.RemoveLayer(i);
        }

        if (prototype.SpritePath is {} path)
        {
            if (_resourceCache.TryGetResource(TextureRoot / path, out RSIResource? resource))
            {
                sprite.BaseRSI = resource.RSI;
            }
        }

        var bodyIndex = sprite.AddBlankLayer();
        var lightStatusIndex = sprite.AddBlankLayer();
        var lightIndex = sprite.AddBlankLayer();

        sprite.LayerMapSet(BorgVisualLayers.Body, bodyIndex);
        sprite.LayerMapSet(BorgVisualLayers.LightStatus, lightStatusIndex);
        sprite.LayerMapSet(BorgVisualLayers.Light, lightIndex);

        sprite.LayerSetState(BorgVisualLayers.Body, prototype.SpriteBodyState);
        sprite.LayerSetState(BorgVisualLayers.LightStatus, prototype.SpriteToggleLightState);

        sprite.LayerSetState(BorgVisualLayers.Light, prototype.SpriteNoMindState);
        sprite.LayerSetVisible(BorgVisualLayers.Light, false);

        if (prototype.SpriteBodyMovementState is { } movementState)
        {
            var spriteMovement = EnsureComp<SpriteMovementComponent>(entity);
            spriteMovement.NoMovementLayers.Clear();
            spriteMovement.NoMovementLayers["movement"] = new PrototypeLayerData
            {
                State = prototype.SpriteBodyState,
            };
            spriteMovement.MovementLayers.Clear();
            spriteMovement.MovementLayers["movement"] = new PrototypeLayerData
            {
                State = movementState,
            };
        }
        else
        {
            RemComp<SpriteMovementComponent>(entity);
        }

        // Update chassis mind states if present
        if (TryComp(entity, out BorgChassisComponent? chassis))
        {
            _borgSystem.SetMindStates(
                (entity.Owner, chassis),
                prototype.SpriteHasMindState,
                prototype.SpriteNoMindState);

            if (TryComp(entity, out AppearanceComponent? appearance))
            {
                _appearance.QueueUpdate(entity, appearance);
            }
        }

        base.UpdateEntityAppearance(entity, prototype);
    }
```

Comment (MilonPL) [reply Content.Client/Silicons/Borgs/BorgSwitchableTypeSystem.cs]: also it throws fucking errors anyways related to the movement for janiborg and othersTM anyways idfk

File: Resources/ConfigPresets/DeltaV/deltav.toml [hunk 13]
```diff
@@ -53,3 +58,21 @@ enabled = false # only enabled on periapsis
 eligible_number = 5
 initiator_ghost_requirement = false
 voter_ghost_requirement = false
+
+[atmos]
+space_wind = false # janky as fuck and probably not suitable for mrp
```
Comment (MilonPL) [inline Resources/ConfigPresets/DeltaV/deltav.toml]: also pro this was supposed to be true :trollface: 
space wind fun for everyone involved, why do you think we have sec magboots

**Pull request merged**